### ISSUE TITLE
Strange behavior of Ducktales: The Quest for Gold when flying Launchpad McQuack for the third time

### ISSUE THREAD

**Issue by BridgeHeadland** [joncampbell123/dosbox-x#3846]
### Describe the bug

It is as it says in the issue. Either I don't get to fly at all, from the start of course, or the plane hangs at the very top, after I get hit by a whale or a thundercloud. What could be the cause? DOSBox-X is not the only DOS emulator where I have experienced this, I have experienced the same in DOSBox SVN Daum. I have had the same files from the Ducktales game untouched in my cloud since 2016, without re-uploading the files. I have downloaded that Ducktales game from other places on the internet than where I downloaded it in 2016, to see if the other copies are different, and experience the same from them as well, with other copies of the files, of course. Therefore, I doubt whether the problem lies in the game, but rather in the DOSBox or in the devices where I have played that Ducktales game. The Ducktales game from the cloud has not behaved like this before.
[dosbox-x.zip](https://github.com/joncampbell123/dosbox-x/files/10070063/dosbox-x.zip)

### Steps to reproduce the behaviour

You just download Ducktales: The Quest for Gold from any webside where you find it avaliable for download. Play it until you flying Launchpad McQuack for the third time and see.

### Expected behavior

I expect the game to behave as it is supposed to behave, without crashes or other nonsense.

### What operating system(s) this bug have occurred on?

Windows 10 Home 21H2, Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz   3.40 GHz

### What version(s) of DOSBox-X have this bug?

The latest versions since last year.

### Used configuration

```ini
Check Describe the bug. :)
```

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by rderooy** [comment]
Did you try to run `loadfix -a` before starting the game?

Also, this game is best run with `machine=tandy` as you will get better sound (and the same graphics as when run with EGA).

**Comment by BridgeHeadland** [comment]
@rderooy Loadfix -a before I start the game? Do you mean I should type 'loadfix -a ducktale.exe', then enter?
Yes, I've tried it in Tandy and get the same problem there too. Speaking of Tandy, as you probably know, I miss a feature that forces the Tandy sound in games like Ducktales: The Quest for Gold, when the machine is set to something higher than the Tandy value, which I mean to remember is impossible, even for the near future developer builds.

**Comment by rderooy** [comment]
Either way. If you run `loadfix -a ducktales.exe` the memory reserved will be automatically released when `ducktales.exe` finishes. If you run `loadfix -a` and then `ducktales.exe` the memory reserved by loadfix will not be automatically released. If that matters or not depends if you want to run other programs afterwards in the same DOSBox-X window that require more conventional memory. You can always release the memory manually with `loadfix -f`.

And regarding tandy, I'm fairly certain that I already mentioned to you that it makes **no difference** to things like CPU speed if you set `machine=tandy` or not. Using `machine=tandy` does not even change the cpu type from the default `cputype=auto`, which is actually an i486 type which Tandy 1000 machines never had.

**Comment by BridgeHeadland** [comment]
The same bug also happens when I use loadfix. The launchpad part of the game can also hang if I play the plane part a second time, but that's rare. It tends to hang the third time I fly Lanuchpad.

**Comment by NebularNerd** [comment]
Just testing the live streamed version over here https://archive.org/details/msdos_Disneys_Duck_Tales_-_The_Quest_for_Gold_1990 and I managed to fly the plane on four occasions on two seperate runs (making sure the second set of runs to hit as much as I could) all flights completed without issue. I've got the .zip of the copy they stream on that page so will test in DosBox shortly. You'll need to use the .pdf manual linked in the comments there to pass the protection scheme.

**Comment by BridgeHeadland** [comment]
@NebularNerd Yes, I still have that problem, both versions with and without copy protection. The error seems to be on all the PCs I play this game on (all the PCs have low memory in them), in any DOSBox, so it's probably not the DOSBox that's at fault. Either it's my custom configuration (on all three configuration files) - which you can find at https://github.com/joncampbell123/dosbox-x/issues/3942 as TXT files, the only thing I've changed now is that I switched out the 8.0 value to the auto value in ver on the DOS section - or the error lies in the devices, i.e. the PCs.
